src/blocks/blocks/posts/components/layout/index.js

```diff
@@ -175,8 +175,8 @@ export const PostsMeta = ({ attributes, element, post, author, categories }) =>
 
 		if ( attributes.displayAuthor && undefined !== author ) {
 
-			/* translators: %s Author of the post */
-			postedOn += sprintf( __( 'by %s', 'otter-blocks' ), author.name );
+			/* translators: %1$s is the date, %2$s is the author of the post */
+			postedOn = sprintf( __( '%1$s by %2$s', 'otter-blocks' ), postedOn, author.name );
```

Comment:
If postedOn is empty until now, we will have an empty leading space. We can remove any leading/trailing spaces on postedOn variable.